src/core/search/search.cc

```diff
@@ -45,20 +45,18 @@ AstExpr ParseQuery(std::string_view query, const QueryParams* params) {
 // Represents an either owned or non-owned result set that can be accessed transparently.
 struct IndexResult {
   using DocVec = vector<DocId>;
-  using BorrowedView = variant<const DocVec*, const CompressedSortedSet*>;
+  using BorrowedView =
+      variant<const DocVec*, const BlockList<CompressedSortedSet>*, const BlockList<SortedVector>*>;
 
   IndexResult() : value_{DocVec{}} {
   }
 
-  IndexResult(const CompressedSortedSet* css) : value_{css} {
-    if (css == nullptr)
-      value_ = DocVec{};
-  }
-
   IndexResult(DocVec&& dv) : value_{std::move(dv)} {
   }
 
-  IndexResult(const DocVec* dv) : value_{dv} {
+  template <typename C> IndexResult(const C* container = nullptr) : value_{container} {
+    if (container == nullptr)
```

Comment:
you can write it in the initialization list with ?:

Reply:
True, but I'll have to add two type annotations for the ternary to have the same types 😟

Reply:
if DocVec can be created from container it should work value_(container ? container : DocVec{})